OMF_API/Program.cs

```diff
@@ -1,540 +1,362 @@
-                RunInTryCatch(sendContainers2, action);
-            // sendContainers2(action);
-
-            RunInTryCatch(sendContainers, action);
-            // sendContainers(action);
-
-            if (sendingToOCS)
-                RunInTryCatch(sendNonTimeStampTypes, action);
-            //sendNonTimeStampTypes(action);
-
-            RunInTryCatch(sendFirstDynamicType, action);
-            RunInTryCatch(sendSecondDynamicType, action);
-            RunInTryCatch(sendThirdDynamicType, action);
-            // sendFirstDynamicType(action);
-            // sendSecondDynamicType(action);
-            // sendThirdDynamicType(action);
-
-            if (!sendingToOCS)
+            catch (WebException e)
             {
-                RunInTryCatch(sendFirstStaticType, action);
-                RunInTryCatch(sendSecondStaticType, action);
-                // sendFirstStaticType(action);
-                // sendSecondStaticType(action);
+                using (WebResponse response = e.Response)
+                {
+                    HttpWebResponse httpResponse = (HttpWebResponse)response;
+
+                    // catch 409 errors as they indicate that the Type already exists
+                    if (httpResponse.StatusCode == HttpStatusCode.Conflict)
+                        return "";
+                    throw;
+                }
             }
         }
 
         /// <summary>
-        /// Sends the values to the preconfigured endpoint
+        /// Sends message to the preconfigured omf endpoint
         /// </summary>
-        /// <param name="messageType"></param>
-        /// <param name="dataJson"></param>
-        /// <param name="action"></param>
-        private static void sendValue(string messageType, string dataJson, string action = "create")
+        /// <param name="endpoint">The endpoint to send an OMF message to</param>
+        /// <param name="messageType">The OMF message type</param>
+        /// <param name="dataJson">The message payload in a string format</param>
+        /// <param name="action">The action for the OMF endpoint to conduct</param>
+        public static void sendMessageToOmfEndpoint(Endpoint endpoint, string messageType, string dataJson, string action = "create")
         {
-            WebRequest request = WebRequest.Create(new Uri(omfendpoint));
+            // create a request
+            WebRequest request = WebRequest.Create(new Uri(endpoint.OmfEndpoint));
             request.Method = "post";
 
-            request.Headers.Add("producertoken", producerToken);
+            // add headers to request
             request.Headers.Add("messagetype", messageType);
             request.Headers.Add("action", action);
-            request.Headers.Add("messageformat", "json");
+            request.Headers.Add("messageformat", "JSON");
             request.Headers.Add("omfversion", omfVersion);
-            if (sendingToOCS)
+            if (string.Equals(endpoint.EndpointType, "OCS"))
```

Comment:
Use the `string.Equals()` overload that specifies a case insensitive `StringComparisonType` here and line 327.